Shared/Samples/Browse organization basemaps/BrowseOrganizationBasemapsView.swift

```diff
@@ -0,0 +1,136 @@
+                        model.selectedItem = .init(basemap: basemap)
+                    }
+                    .buttonStyle(.plain)
+                }
+            case .failure:
+                urlEntryView
+                ContentUnavailableView(
+                    "Error",
+                    systemImage: "exclamationmark.triangle",
+                    description: Text("Error searching specified portal.")
+                )
+            case nil:
+                if model.isConnecting {
+                    ProgressView()
+                        .frame(maxWidth: .infinity)
+                } else {
+                    urlEntryView
+                }
+            }
+        }
+        .sheet(item: $model.selectedItem) { selectedItem in
+            NavigationStack {
+                MapView(map: Map(basemap: selectedItem.basemap))
+                    .navigationTitle(selectedItem.basemap.title)
+                    .navigationBarTitleDisplayMode(.inline)
+                    .toolbar {
+                        ToolbarItem(placement: .topBarTrailing) {
+                            Button("Done") {
+                                model.selectedItem = nil
+                            }
+                        }
+                    }
+            }
+            .interactiveDismissDisabled()
+            .highPriorityGesture(DragGesture())
+            .pagePresentation()
+        }
+        .animation(.default, value: model.isConnecting)
+    }
+    
+    @ViewBuilder private var urlEntryView: some View {
+        Section {
+            HStack {
+                TextField("Portal URL", text: $model.portalURLString)
+                    .onSubmit { Task { await model.connectToPortal() } }
+                    .autocapitalization(.none)
+                    .keyboardType(.URL)
+                Button("Connect") {
+                    Task { await model.connectToPortal() }
+                }
+                .disabled(model.portalURL == nil)
+            }
+        }
+    }
+}
+
+/// A value that represents an item selected by the user.
+private struct SelectedItem: Identifiable {
+    /// The basemap was selected.
+    let basemap: Basemap
+    
+    var id: ObjectIdentifier {
+        ObjectIdentifier(basemap)
+    }
+}
+
+private extension Basemap {
+    /// The title of the item, or "unknown" when the `item` is `nil`.
+    var title: String { item?.title ?? "unknown" }
+}
+
+extension BrowseOrganizationBasemapsView {
+    @MainActor
+    class Model: ObservableObject {
+        /// The URL string entered by the user.
+        @Published var portalURLString = ""
+        
+        /// The fetched portal content.
+        @Published var basemaps: Result<[Basemap], Error>?
+        
+        /// A Boolean value indicating if a portal connection is in progress.
+        @Published var isConnecting = false
+        
+        /// The selected item.
+        @Published fileprivate var selectedItem: SelectedItem?
```

Comment:
```suggestion
        @Published private var selectedItem: SelectedItem?
```
Does this need to be `fileprivate`?

Reply:
yes because SelectedItem is private